server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowTaskInstanceService.java

```diff
@@ -192,33 +180,43 @@ public FlowInstanceDetailResp executeTask(@NotNull Long id) throws IOException {
         return FlowInstanceDetailResp.withIdAndType(id, taskInstance.getTaskType());
     }
 
-    public String getLog(@NotNull Long id, OdcTaskLogLevel level) throws IOException {
-        Optional<TaskEntity> taskEntityOptional = getTaskEntity(id,
-                instance -> (instance.getStatus().isFinalStatus() || instance.getStatus() == FlowNodeStatus.EXECUTING)
-                        && instance.getTaskType() != TaskType.SQL_CHECK
-                        && instance.getTaskType() != TaskType.PRE_CHECK
-                        && instance.getTaskType() != TaskType.GENERATE_ROLLBACK);
+    public String getLog(@NotNull Long flowInstanceId, OdcTaskLogLevel level) throws IOException {
+        Optional<TaskEntity> taskEntityOptional = getLogDownloadableTaskEntity(flowInstanceId);
         if (!taskEntityOptional.isPresent()) {
             return null;
         }
         TaskEntity taskEntity = taskEntityOptional.get();
-        if (taskEntity.getResultJson() == null) {
-            return null;
+        if (!dispatchChecker.isTaskEntityOnThisMachine(taskEntity)) {
```

Comment:
why should we check this first?
we may check if this task is impled by task framework, should it be better?

Reply:
Whether enabled task framework is not matter, it is just a different way for finding the log content. What we firstly need to do is determine whether the request is solved by the right meachine.

Reply:
Yes you are right, done!